src/Symfony/Component/Notifier/Bridge/FreeMobile/FreeMobileTransport.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\FreeMobile;
+
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Antoine Makdessi <amakdessi@me.com>
+ *
+ * @experimental in 5.1
+ */
+final class FreeMobileTransport extends AbstractTransport
+{
+    protected const HOST = 'https://smsapi.free-mobile.fr/sendmsg';
+
+    private $login;
+    private $password;
+
+    public function __construct(string $login, string $password, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->login = $login;
+        $this->password = $password;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('freemobile://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
```

Comment:
What about check an additional information about the `$message` as the payload is not exactly the same as others?

Reply:
@Guikingone what do you mean?

like checking the sms phone is empty as it will be not used?

Reply:
By example but I'm more worried about the fact that the TwilioTransport use the same approach to detect if it can support `SmsMessage`: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Notifier/Bridge/Twilio/TwilioTransport.php

I don't know if it's possible but it could be a good idea to differentiate both transport when it comes to detect if it can support the message ? 🤔

Reply:
It is the same implementation for Nexmo as well, so maybe there is another thing that differentiate the notifier event if it is « supported »
I did not look at the internal component code yet tbh

Reply:
The idea is that we can choose between providers that support sending SMS messages.

Here, the provider is a bit special as you can only send to your number. I think we need to add this phone number in the DSN that is then used here in the `supports` method to trigger this Transport. It's then your responsibility to make this transport the first one. That way, if you are using the special number, the Free notifier is used, if not, it falls back to other SMS providers. And in any case, the SMS message can also be send by other providers if needed.

Reply:
ok I see, I will improve this one 
thank you

Reply:
@fabpot @Guikingone I've tried what you suggested, but I do not know if it is what both of you have in mind
also I added some test based on other transports